lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +37,92 @@ describe('NxModal', function() {
+    let containerMainModal: HTMLDivElement | null;
+    let containerSecondaryModal: HTMLDivElement | null;
+
+    beforeEach(function () {
+      // Rendering containerMainModal for the component in test.
+      containerMainModal = document.createElement('div');
+      containerSecondaryModal = document.createElement('div');
+
+      document.body.appendChild(containerMainModal);
+      document.body.appendChild(containerSecondaryModal);
+    });
+
+    afterEach(function () {
+      if (containerMainModal) {
+        document.body.removeChild(containerMainModal);
+        containerMainModal = null;
+      }
+
+      if (containerSecondaryModal) {
+        document.body.removeChild(containerSecondaryModal);
+        containerSecondaryModal = null;
+      }
+    });
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = <NxModal id="first-modal-id" onClose={mockCallBack}/>;
+
+      act(() => {
+        mount(nxModal, {attachTo: containerMainModal});
+      });
+
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(mockCallBack).toHaveBeenCalledTimes(1);
+    });
+
+    it('executes onClose method for each modal that has been opened, preserving order call', function () {
+      const firstMockCallBack = jest.fn();
+      const secondMockCallBack = jest.fn();
+      const firstNxModal = <NxModal id="first-modal-id" onClose={firstMockCallBack}/>;
+      const secondNxModal = <NxModal id="second-modal-id" onClose={secondMockCallBack}/>;
+
+      let secondModalWrapper: ReactWrapper | null = null;
+      act(() => {
+        mount(firstNxModal, {attachTo: containerMainModal});
+        secondModalWrapper = mount(secondNxModal, {attachTo: containerSecondaryModal});
+      });
+
+      // "Close" the second modal
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(1);
+      expect(firstMockCallBack).not.toHaveBeenCalled();
+
+      // Unmount and remount the modal to ensure listener order is preserved if new modals are opened
+      act(() => {
+        secondModalWrapper!.unmount();
+        secondModalWrapper = mount(secondNxModal, {attachTo: containerSecondaryModal});
+      });
+      // "Close" the second modal again
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(2);
+      expect(firstMockCallBack).not.toHaveBeenCalled();
+
+      // Unmount second modal to remove it's listener from available listeners to execute
+      act(() => {
+        secondModalWrapper!.unmount();
+      });
+      // "Close" the first modal
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(secondMockCallBack).toHaveBeenCalledTimes(2);
+      expect(firstMockCallBack).toHaveBeenCalledTimes(1);
+    });
```

Comment:
I'd add one more esc press at the end here to assert that it doesn't call either callback an additional time when all the modals are already closed.

Reply:
> I'd add one more esc press at the end here to assert that it doesn't call either callback an additional time when all the modals are already closed.

Additional assert for handler execution: 110be86